src/Symfony/Component/Yaml/Dumper.php

```diff
@@ -84,6 +84,30 @@ public function dump($input, $inline = 0, $indent = 0, $flags = 0)
             $isAHash = array_keys($input) !== range(0, count($input) - 1);
 
             foreach ($input as $key => $value) {
+                if ($inline - 1 > 0 && Yaml::DUMP_MULTI_LINE_AS_BLOCK & $flags && is_string($value) && false !== strpos($value, "\n")) {
+                    $output .= sprintf(
+                        "%s%s%s |\n",
+                        $prefix,
+                        $isAHash ? Inline::dump($key, $flags).':' : '-',
+                        ''
+                    );
+
+                    foreach (preg_split("/\n|\r\n/", $value) as $row) {
```

Comment:
I think this should use a single-quoted string, to put ``\n`` in the regex instead of putting LF. This is safer to tell the regex engine that we want a newline char rather than expecting that it treats multiline regexes as newlines to be matched IMO